demisto_sdk/commands/common/tools.py

```diff
@@ -2591,15 +2591,15 @@ def get_api_module_dependencies(pkgs, id_set_path, verbose):
     """
 
     id_set = open_id_set_file(id_set_path)
-    api_modules = [pkg.name for pkg in pkgs if API_MODULES_PACK in pkg.parts]
+    api_modules = {pkg.name for pkg in pkgs if API_MODULES_PACK in pkg.parts}
     scripts = id_set.get(IdSetKeys.SCRIPTS.value, [])
     integrations = id_set.get(IdSetKeys.INTEGRATIONS.value, [])
     using_scripts, using_integrations = [], []
     for script in scripts:
         script_info = list(script.values())[0]
         script_name = script_info.get('name')
         api_module = script_info.get('api_modules', [])
-        if api_module in api_modules:
+        if len(api_modules.intersection(api_module)) > 0:
```

Comment:
```suggestion
        if api_modules & api_module:
```

Reply:
@dorschw `api_module` is a list and `api_modules` is a set. First - @JudahSchwartz I think we should make the names clearer i.e. change `api_module` to `script_ api_modules` and `api_modules` to `changed_ api_modules`. Then I am not sure what you (@dorschw ) will work and the intersection will be necessary here.

Reply:
Also in the next loop